src/dataregistry/registrar_util.py

```diff
@@ -49,3 +49,34 @@ def form_dataset_path(owner_type, owner, relative_path, root_dir=None):
     if root_dir:
         to_return = os.path.join(root_dir, to_return)
     return to_return
+
+def get_directory_info(path):
+    """
+    Get the total disk space used by a directory and the total number of files
+    in the directory (includes subdirectories):
+
+    Parameters
+    ----------
+    path : str
+        Location of directory
+
+    Returns
+    -------
+    num_files : int
+        Total number of files in dir (including subdirectories)
+    total_size : float
+        Total disk space (in bytes) used by directory (including subdirectories)
+    """
+
+    num_files = 0
+    total_size = 0
+    with os.scandir(path) as it:
+        for entry in it:
+            if entry.is_file():
+                num_files += 1
+                total_size += entry.stat().st_size
```

Comment:
I suspect calling `stat( )` on every file will be rather slow.      It looks like `shutil.disk_usage` will do this for you.  It probably calls `stat( )` repeatedly itself, but that should be significantly faster than doing it one by one in python.
I don't know of a way to get counts so simply, but if that's all you're having to do in this code it might be possible to streamline somewhat.

Reply:
`shutil.disk_usage` gives you info about the disk the directory is on, not the directory itself (i.e., its doing `df /path` rather than `du /path`).

i dont think theres a away around doing a loop.

ive added some verbosity (with a flag that can switch it off), that will give us an idea of how long the metadata probing, and copying takes

Reply:
will leave it like this for now, will keep track of it through an issue to see if/can speed it up later